up-l1/someip.adoc

```diff
@@ -0,0 +1,221 @@
+
+| `UMESSAGE_TYPE_REQUEST` |`REQUEST` |Requests type message
+
+| `UMESSAGE_TYPE_RESPONSE` |`RESPONSE` or `ERROR` |Responses or error has occurred while attempting to deliver the message or a service
+has thrown an exception
+
+|===
+
+
+When receiving uProtocol initiated requests:
+
+* *MUST* cache the request `UAttributes` for a maximum of `ttl` so that it can be used to build a response `UAttributes` when receiving a response from SOME/IP
+
+```
+response.priority = request.priority
+response.reqid = request.id
+```
+
+* Auto-generated SOME/IP REQUEST message, *MAY* set the 16 bit client ID to the 0 per SOME/IP specifications 
+
+When receiving a SOME/IP initiated requests:
+
+* *MUST* cache the SOME/IP request ID as well as the generated `UAttributes` for the request messages so that the response can be translated back to a SOME/IP response message
+
+
+==== Communication Status 
+
+<<commstatus-mapping>> below provides the mapping of uAttributes `commstatus` `UCode` codes to SOME/IP error codes.
+
+.Error Code Mapping
+[commstatus-mapping, width=70%]
+|===
+| UCode | SOME/IP Error Codes
+
+|`OK` |`E_OK`
+|`INVALID_ARGUMENT` |`E_WRONG_MESSAGE_TYPE` or `E_UNKNOWN_METHOD`
+|`DEADLINE_EXCEEDED` |`E_TIMEOUT`
+|`NOT_FOUND` |`E_UNKNOWN_SERVICE`
+|`UNAVAILABLE` |`E_NOT_READY`
+|`DATA_LOSS` |`E_MALFORMED_MESSAGE`
+|`INTERNAL` |`E_NOT_REACHABLE`
+|===
+
+
+=== UPayload
+
+`UPayload` hosts the application layer data that is being sent between devices. `UPayload` structure, defined in link:../basics/upayload.adoc[UPayload] contains either a copy or reference to the actual payload that is to be sent. `UPayload` also contains `UPayloadFormat` used to give a hint of the payload format (protobuf serialized, SOME/IP format, TEXT, RAW, etc...). The SOME/IP specifications however does not have an equivalent field for `UPayloadFormat` as it is assumed that the payload is serialized in the format that the other end knows how to deserialize (i.e. it is fixed per topic). As such, when converting between uProtocol and SOME/IP, the `UPayloadFormat` field *SHOULD* be ignored (left at the default of `UMESSAGE_TYPE_UNSPECIFIED`).
+
+
+=== uTransport
+
+Mapping of uTransport APIs to SOME/IP specific library APIs shall not be covered in this document given there are multiple open source libraries available for SOME/IP.
+
+
+== Application Layer Translation
+
+Application (or message payload) translation is the process of converting SOME/IP-SD subscription and discovery messages, to/from uDiscovery and uSubscription Messages. 
+
+=== uSubscription
+
+The following section will elaborate only on the translation of uSubscription messages to/from SOME/IP-SD messages. Subscription state (persistent or not) is handled in the uSubscription services and not at the transport layer or this component.
+
+The following section we will elaborate on how Eventgroup Entry types (5.1.3.1) are mapped to uSubscription messages for the subscribe
```

Comment:
As it was harder to find this reference, can you also mention `[PRS_SOMEIPSD_00385]`